src/org/labkey/test/components/ui/grids/GridBar.java

```diff
@@ -356,8 +362,8 @@ else if(onSamplePage)
                 break;
         }
 
-        doMenuAction(currentButtonText, Arrays.asList(menuChoice));
-
+        String finalMenuChoice = menuChoice;
+        _queryGrid.doAndWaitForUpdate(()->doMenuAction(currentButtonText, Arrays.asList(finalMenuChoice)));
```

Comment:
Creating this extra `final` variable isn't necessary. We could make `menuChoice` final by removing its initial assignment and adding a `default` branch to the view switch.

Reply:
Changed.